src/use-built-ins-plugin.js

```diff
@@ -0,0 +1,227 @@
+import { definitions } from "./built-in-definitions";
+
+function isPolyfillSource(value) {
+  return value === "babel-polyfill";
+}
+
+function warnOnInstanceMethod(details) {
+  console.warn(
+    `Adding a polyfill: An instance method may have been used: ${details}`,
+  );
+}
+
+function has(obj, key) {
+  return Object.prototype.hasOwnProperty.call(obj, key);
+}
+
+function getObjectString(node) {
+  if (node.type === "Identifier") {
+    return node.name;
+  } else if (node.type === "MemberExpression") {
+    return `${getObjectString(node.object)}.${getObjectString(node.property)}`;
+  } else {
+    return "";
+  }
+}
+
+export default function({ types: t }) {
+  function addImport(path, builtIn, builtIns) {
+    if (builtIn && !builtIns.has(builtIn)) {
+      builtIns.add(builtIn);
+      const importDec = t.importDeclaration([], t.stringLiteral(builtIn));
+      importDec._blockHoist = 3;
+      const programPath = path.find(path => path.isProgram());
+      programPath.unshiftContainer("body", importDec);
+    }
+  }
+
+  function addUnsupported(path, polyfills, builtIn, builtIns) {
+    if (Array.isArray(builtIn)) {
+      for (const i of builtIn) {
+        if (polyfills.indexOf(i) !== -1) {
```

Comment:
Maybe polyfills could be a `Set` so you get O(1) lookups instead of O(n)? Might be premature optimization.